From 12324fd6ff0e806d6c246c1ce48685060c9e1c29 Mon Sep 17 00:00:00 2001
From: George Zacharia <george.zcharia@gmail.com>
Date: Fri, 17 Jan 2020 10:10:45 +0530
Subject: [PATCH] Add a proper FAB

Also hide the FAB on scroll
Looks good and makes it easier for glancing through messages

TODO: Use a extended FAB when Material themes are implemented
---
 .../java/com/moez/QKSMS/feature/main/MainActivity.kt | 11 +++++++++++
 presentation/src/main/res/layout/main_activity.xml   | 12 ++++++------
 2 files changed, 17 insertions(+), 6 deletions(-)

diff --git a/presentation/src/main/java/com/moez/QKSMS/feature/main/MainActivity.kt b/presentation/src/main/java/com/moez/QKSMS/feature/main/MainActivity.kt
index 362f66a6..a73ecc08 100644
--- a/presentation/src/main/java/com/moez/QKSMS/feature/main/MainActivity.kt
+++ b/presentation/src/main/java/com/moez/QKSMS/feature/main/MainActivity.kt
@@ -38,6 +38,7 @@ import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.ViewModelProviders
 import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.snackbar.Snackbar
 import com.jakewharton.rxbinding2.view.clicks
 import com.jakewharton.rxbinding2.widget.textChanges
@@ -259,6 +260,16 @@ class MainActivity : QkThemedActivity(), MainView {
                 conversationsAdapter.updateData(state.page.data)
                 itemTouchHelper.attachToRecyclerView(recyclerView)
                 empty.setText(R.string.inbox_empty_text)
+
+                recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+                    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                        if (dy > 0 && compose.visibility == View.VISIBLE) {
+                            compose.hide()
+                        } else if (dy < 0 && compose.visibility != View.VISIBLE) {
+                            compose.show()
+                        }
+                    }
+                })
             }
 
             is Searching -> {
diff --git a/presentation/src/main/res/layout/main_activity.xml b/presentation/src/main/res/layout/main_activity.xml
index 4b9bebf2..5cb91caf 100644
--- a/presentation/src/main/res/layout/main_activity.xml
+++ b/presentation/src/main/res/layout/main_activity.xml
@@ -91,16 +91,16 @@
             app:layout_constraintTop_toBottomOf="@id/toolbar"
             tools:listitem="@layout/conversation_list_item" />
 
-        <ImageView
+        <com.google.android.material.floatingactionbutton.FloatingActionButton
             android:id="@+id/compose"
-            android:layout_width="56dp"
-            android:layout_height="56dp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            style="@style/Widget.MaterialComponents.FloatingActionButton"
+            android:layout_gravity="bottom|end"
             android:layout_margin="16dp"
             android:background="@drawable/circle_ripple"
             android:contentDescription="@string/main_fab_cd"
-            android:elevation="8dp"
-            android:padding="16dp"
-            android:src="@drawable/ic_add_black_24dp"
+            app:srcCompat="@drawable/ic_add_black_24dp"
             android:tint="@color/white"
             app:layout_constraintBottom_toTopOf="@id/bottom"
             app:layout_constraintEnd_toEndOf="parent"
